planner/core/testdata/index_merge_suite_out.json

```diff
@@ -49,6 +49,128 @@
           "  └─Selection(Probe) 0.00 cop[tikv]  lt(test.t.a, 10)",
           "    └─TableRowIDScan 0.00 cop[tikv] table:t keep order:false, stats:pseudo"
         ]
+      },
+      {
+        "SQL": "select /*+ use_index_merge(t, j0_0) */ * from t where json_contains((j0->'$.path0'), '[1, 2, 3]')",
+        "Plan": [
+          "IndexMerge 0.00 root  type: intersection",
+          "├─IndexRangeScan(Build) 10.00 cop[tikv] table:t, index:j0_0(cast(json_extract(`j0`, _utf8mb4'$.path0') as signed array)) range:[1,1], keep order:false, stats:pseudo",
+          "├─IndexRangeScan(Build) 10.00 cop[tikv] table:t, index:j0_0(cast(json_extract(`j0`, _utf8mb4'$.path0') as signed array)) range:[2,2], keep order:false, stats:pseudo",
+          "├─IndexRangeScan(Build) 10.00 cop[tikv] table:t, index:j0_0(cast(json_extract(`j0`, _utf8mb4'$.path0') as signed array)) range:[3,3], keep order:false, stats:pseudo",
+          "└─TableRowIDScan(Probe) 0.00 cop[tikv] table:t keep order:false, stats:pseudo"
+        ]
+      },
+      {
+        "SQL": "select /*+ use_index_merge(t, j0_0) */ * from t where json_overlaps((j0->'$.path0'), '[1, 2, 3]')",
+        "Plan": [
+          "Selection 0.00 root  json_overlaps(json_extract(test.t.j0, \"$.path0\"), cast(\"[1, 2, 3]\", json BINARY))",
```

Comment:
It looks like this `Selection` is redundant.

Reply:
The reason is `ast.JSONOverlaps` cannot be pushed down to TiKV (https://github.com/pingcap/tidb/blob/master/expression/expression.go#L1036), and this let the optimizer keeps an Selection upon the DataSrouce (https://github.com/pingcap/tidb/blob/master/planner/core/rule_predicate_push_down.go#L141), which is expected.

Reply:
But the `json_overlaps` has already been handled by the `IndexMerge`.
And `json_contains` also doesn't have the `Selection`.

Reply:
YES, but it's another problem. I'll submit another PR to fix it after.